packages/calcite-components/src/components/rating/rating.tsx

```diff
@@ -407,35 +353,77 @@ export class Rating
       } else if (this.required && numberKey > 0 && numberKey <= this.max) {
         this.value = numberKey;
       }
+      this.updatefocus();
     }
   };
 
   private handleInputChange = (event: InputEvent) => {
     if (this.isKeyboardInteraction === true) {
       const inputVal = Number(event.target["value"]);
-      this.focusValue = inputVal;
       this.hoverValue = inputVal;
       this.value = inputVal;
     }
   };
 
   private handleLabelPointerOver = (event: PointerEvent) => {
-    const target = event.currentTarget as HTMLLabelElement;
-    const newPointerValue = Number(target.firstChild["value"] || 0);
-    this.hoverValue = newPointerValue;
-    this.focusValue = null;
+    this.hoverValue = this.getValueFromLabelEvent(event);
   };
 
   private handleLabelPointerDown = (event: PointerEvent) => {
     const target = event.currentTarget as HTMLLabelElement;
-    const inputVal = Number(target.firstChild["value"] || 0);
+    const inputValue = Number(target.firstChild["value"]);
+    this.hoverValue = inputValue;
+    this.emit = true;
+    this.value = !this.required && this.value === inputValue ? 0 : inputValue;
+    target.focus();
+  };
 
-    this.focusValue = null;
+  private handleLabelClick = (event: Event) => {
+    event.preventDefault();
+  };
+
+  private handleLabelFocus = (event: FocusEvent) => {
+    const inputValue = this.getValueFromLabelEvent(event);
+    this.hoverValue = inputValue;
+  };
+
+  private handleHostBlur = () => {
     this.hoverValue = null;
-    this.emit = true;
-    this.value = !this.required && this.value === inputVal ? 0 : inputVal;
   };
 
+  private updatefocus(): void {
+    this.hoverValue = this.value;
+    this.labelElements[this.value - 1].focus();
+  }
+
+  private getTabIndex(value: number): number {
+    if (this.readOnly) {
+      return -1;
+    }
+    if (this.value) {
+      return this.value === value ? 0 : -1;
+    } else {
+      return value === 1 ? 0 : -1;
+    }
+  }
+
+  private setLabelEl = (el: HTMLLabelElement): void => {
+    this.labelElements.push(el);
+  };
+
+  private getValueFromLabelEvent(event: FocusEvent | PointerEvent | KeyboardEvent): number {
+    const target = event.currentTarget as HTMLLabelElement;
+    return Number(target.firstChild["value"]) || 0;
```

Comment:
maybe instead of using `target.firstChild` you can use the `data-value` attribute?